### ISSUE TITLE
Generic enum in @MappedSuperclass fails in Quarkus 3.0.0.CR2

### ISSUE THREAD

**Issue by mensinda** [quarkusio/quarkus#32664]
### Describe the bug

Using a generic type in an abstract `@MappedSuperclass` results in an `java.lang.IllegalArgumentException: Named type [interface root.EnumBase] did not implement BasicType nor UserType` exception on statup.

This "used to work" in Quarkus 2.x / Hibernate 5.x, where at least the `SessionFactory` was built. However, there seems to be an issue with the dirty check when **only** the property of the `@MappedSuperclass` is updated ~~via **methods of the  `@MappedSuperclass`** but *not* methods of the child class~~.

### Expected behavior

All tests pass in Quarkus 2.x and 3.x (Hibernate 5.x, 6.x)

#### Quarkus 3.x / Hibernate 6.x

The test throws the following exception:

```
java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:625)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:696)

	// More junit lines that are probably irrelavent

	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:50)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:273)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:250)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:592)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:640)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$12(ClassBasedTestDescriptor.java:395)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:395)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:211)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:84)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
	... 37 more
Caused by: java.lang.RuntimeException: jakarta.persistence.PersistenceException: [PersistenceUnit: <default>] Unable to build Hibernate SessionFactory
	at io.quarkus.hibernate.orm.runtime.JPAConfig.startAll(JPAConfig.java:78)
	at io.quarkus.hibernate.orm.runtime.HibernateOrmRecorder.startAllPersistenceUnits(HibernateOrmRecorder.java:101)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits1868654632.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits1868654632.deploy(Unknown Source)
	... 53 more
Caused by: jakarta.persistence.PersistenceException: [PersistenceUnit: <default>] Unable to build Hibernate SessionFactory
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.persistenceException(FastBootEntityManagerFactoryBuilder.java:121)
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:81)
	at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:74)
	at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
	at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
	at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:156)
	at io.quarkus.hibernate.orm.runtime.JPAConfig$1.run(JPAConfig.java:64)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalArgumentException: Named type [interface root.EnumBase] did not implement BasicType nor UserType
	at org.hibernate.boot.model.TypeDefinition.createResolution(TypeDefinition.java:277)
	at org.hibernate.boot.model.TypeDefinition.createResolution(TypeDefinition.java:123)
	at org.hibernate.boot.model.TypeDefinition.resolve(TypeDefinition.java:108)
	at org.hibernate.mapping.BasicValue.interpretExplicitlyNamedType(BasicValue.java:663)
	at org.hibernate.mapping.BasicValue.buildResolution(BasicValue.java:374)
	at org.hibernate.mapping.BasicValue.resolve(BasicValue.java:310)
	at org.hibernate.mapping.BasicValue.getType(BasicValue.java:288)
	at org.hibernate.metamodel.internal.AttributeFactory.determineAttributeMetadata(AttributeFactory.java:407)
	at org.hibernate.metamodel.internal.AttributeFactory.buildAttribute(AttributeFactory.java:115)
	at org.hibernate.metamodel.internal.AttributeFactory.buildAttribute(AttributeFactory.java:100)
	at org.hibernate.metamodel.internal.MetadataContext.wrapUp(MetadataContext.java:332)
	at org.hibernate.metamodel.model.domain.internal.JpaMetamodelImpl.processJpa(JpaMetamodelImpl.java:531)
	at org.hibernate.metamodel.model.domain.internal.MappingMetamodelImpl.finishInitialization(MappingMetamodelImpl.java:213)
	at org.hibernate.internal.SessionFactoryImpl.initializeMappingModel(SessionFactoryImpl.java:319)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:269)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:200)
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:79)
	... 6 more
```

#### Quarkus 2.x / Hibernate 5.x

Here, the 

### How to Reproduce?

To reproduce the error:

1. Check out the `genericTests` branch of this repository: https://github.com/mensinda/quarkus-stuff/tree/genericTests
2. run `mvn clean verify`

To see that it "used to work" in Quarkus 2.x:

1. run `./downgrade.sh`
2. run `mvn clean verify`

### Output of `uname -a` or `ver`

Linux 5.15.0-56-generic #62-Ubuntu SMP Tue Nov 22 19:54:14 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.6" 2023-01-17

### Quarkus version or git rev

3.0.0.CR2 / 2.16.6.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.2 (ea98e05a04480131370aa0c110b8c54cf726c06f)

**Comment by gsmet** [comment]
@beikov @dreab8 could one of you have a look at this one while Yoann is on PTO? There is a reproducer in the description (in a branch of the pointed repository).

Thanks!

**Comment by dreab8** [comment]
Hi @gsmet, I'm going to give a look at it

**Comment by dreab8** [comment]
Hi @gsmet , created https://hibernate.atlassian.net/browse/HHH-16479

**Comment by yrodiere** [comment]
Fixed by #33498
